### PR TITLE
DEP: Update pandas min version to 1.1

### PR DESCRIPTION
**Pull request by m-richards**
I was coming back to look at #2542 to review and noted the CI failure there is only for pandas 1.0.5. In in another issue it was noted that this actually falls outside the supported depency range we follow based on #1457.

So this bumps the minmum pandas version to 1.1 and it also seemed sensible to look at the minimum version of our other dependencies.

(Perhaps it would make more sense to do this just after a release date is approximately locked in, or after a release - so I'm happy to revert changes to the other packages and keep it just to pandas for now if that makes more sense?)

Thus far I've bumped the following minimum versions:
- Shapely 1.7.1 (released 2020-08-20) - 1.7 was released January 2020 (technically based on supporting dependencies released in the last two years as suggested in NEP29 this should be shapely 1.8.0, but since this was released nearly a year later in Oct 2021, I've opted to keep 1.7.1 for now)
- Numpy 1.20 - NEP 29 says support for 1.19 dropped on Jun 21, 2022
- Fiona 1.8.19 (released Apr 2, 2021 and 1.8.18 before released on Nov 18 2020 and so is outside the 2 year support window)
- PyProj 3.0.1 (released Mar 5, 2021 and 3.0.0.post1 before was released Nov 6, 2020 so is outside the window)
- Matplotlib 3.3.4 (released Jan 28, 2021, 3.3.3 released Nov 12, 2020)

(Note at this point I still haven't updated the pyproject.toml, docs, etc, I'll see if I've broken the CI before doing this.

### CODE REVIEW

**Comment by rraymondgh** [comment]
I have actually been looking at this systematically.  Getting release dates from **github** and then comparing to compliance with https://numpy.org/neps/nep-0029-deprecation_policy.html

I think min versions become:

| package     |     min |     mid |     max | min_date            | mid_date            | max_date            |
|:------------|--------:|--------:|--------:|:--------------------|:--------------------|:--------------------|
| fiona       |    1.8  |    1.8  |    1.8  | 2022-02-07 00:00:00 | 2022-02-07 00:00:00 | 2022-02-07 00:00:00 |
| folium      |    0.12 |    0.13 |    0.13 | 2021-01-04 00:00:00 | 2022-10-07 00:00:00 | 2022-10-07 00:00:00 |
| geoalchemy2 |    0.1  |    0.11 |    0.12 | 2021-12-24 00:00:00 | 2022-03-04 00:00:00 | 2022-06-07 00:00:00 |
| geopandas   |    0.8  |    0.1  |    0.12 | 2021-01-25 00:00:00 | 2021-10-03 00:00:00 | 2022-10-24 00:00:00 |
| geopy       |    2.1  |    2.2  |    2.3  | 2020-12-27 00:00:00 | 2021-07-11 00:00:00 | 2022-11-13 00:00:00 |
| mapclassify |    2.4  |    2.4  |    2.4  | 2020-12-14 00:00:00 | 2020-12-14 00:00:00 | 2020-12-14 00:00:00 |
| matplotlib  |    3.3  |    3.5  |    3.6  | 2020-11-12 00:00:00 | 2021-11-16 00:00:00 | 2022-09-16 00:00:00 |
| numpy       |    1.19 |    1.21 |    1.23 | 2020-10-29 00:00:00 | 2021-06-19 00:00:00 | 2022-06-22 00:00:00 |
| packaging   |   20.5  |   20.9  |   21.3  | 2020-11-27 00:00:00 | 2021-01-29 00:00:00 | 2021-11-18 00:00:00 |
| pandas      |    1.1  |    1.3  |    1.5  | 2020-10-30 00:00:00 | 2021-07-02 00:00:00 | 2022-09-19 00:00:00 |
| pygeos      |    0.9  |    0.11 |    0.13 | 2021-01-23 00:00:00 | 2021-10-30 00:00:00 | 2022-08-25 00:00:00 |
| pyogrio     |    0.2  |    0.3  |    0.4  | 2021-04-01 00:00:00 | 2021-12-22 00:00:00 | 2022-06-20 00:00:00 |
| pyproj      |    3    |    3.2  |    3.4  | 2020-11-05 00:00:00 | 2021-09-04 00:00:00 | 2022-09-10 00:00:00 |
| python      |    3.7  |    3.9  |    3.11 | 2018-06-27 00:00:00 | 2020-10-05 00:00:00 | 2022-10-24 00:00:00 |
| rtree       |    0.9  |    1    |    1    | 2020-12-16 00:00:00 | 2022-04-04 00:00:00 | 2022-04-04 00:00:00 |
| scipy       |    1.5  |    1.7  |    1.9  | 2020-11-04 00:00:00 | 2021-06-19 00:00:00 | 2022-07-29 00:00:00 |
| shapely     |    1.8  |    1.8  |    1.8  | 2021-10-21 00:00:00 | 2021-10-21 00:00:00 | 2021-10-21 00:00:00 |
| sqlalchemy  |    1.3  |    1.4  |    1.4  | 2020-12-17 00:00:00 | 2021-03-15 00:00:00 | 2021-03-15 00:00:00 |
| xyzservices | 2021.7  | 2022.1  | 2022.9  | 2021-07-30 00:00:00 | 2022-01-17 00:00:00 | 2022-09-19 00:00:00 |

I have a jupyter notebook that extracts and generates this (that I can't attach!).  I'm looking to refresh all the environment files in a re-produceable way which can be re-run periodically (maybe on a release basis).  Also getting the whole framework to run locally using https://github.com/nektos/act

**Comment by m-richards** [comment]
> I have actually been looking at this systematically. Getting release dates from **github** and then comparing to compliance with https://numpy.org/neps/nep-0029-deprecation_policy.html

I think this could be quite useful - as this is something that we periodically should review. I haven't looked to see if there is already a tool for this (the above I just did manually as I was looking at the package versionings which are pinned in the CI environments. I don't think this is automatable though - for instance take fiona - it's missing releases on github, but they are there on pypi. Additionally we'd want to make sure only to change the dependencies which actually matter, we don't do any version checks for e.g. folium, geopy, packaging, scipy in the geopandas code, so there's no benefit to pinning them.

> I'm looking to refresh all the environment files in a re-produceable way which can be re-run periodically (maybe on a release basis). Also getting the whole framework to run locally using https://github.com/nektos/act

This could also be useful (for instance I updated a numpy pin in one of the CI files and it wasn't clear what version it would make the most sense to update to). But on the other hand sometimes we need to test versions in specific situations rather than based on a schedule - for instance if there are known regressions.
 
> Probably a separate issue / PR. I think **ci** should be `pytest -q` instead of `pytest -v`. The verbose output just hides the warnings and errors. It's far easier to get to wanted content without the verbose logging.

A separate issue to discuss would make sense.

**Comment by martinfleis** [comment]
The pins look okay to me. It seems that some work on optional pins needs to be done to make CI green.

**Comment by m-richards** [comment]
alright, other ci envs are now green. For reasons I don't quite understand I had to lower fiona to 1.8.18 (see this failure referencing orc and protobuf). Without forcing pyarrow=8, I was able to get an environment with fiona 1.8.19 to resolve, but that ended up installing pyarrow 3, which fails our tests. For now I have only lowered it in the ci, and not in docs/ other places.

File: geopandas/tools/tests/test_sjoin.py [hunk 1]
```diff
@@ -365,9 +364,7 @@ def test_right(self, predicate, dfs):
             )
             exp.index.names = df2.index.names
 
-        # GH 1364 fix of behaviour was done in pandas 1.1.0
-        if predicate == "within" and Version(pd.__version__) >= Version("1.1.0"):
-            exp = exp.sort_index()
+        exp = exp.sort_index()
```
Comment (martinfleis) [inline geopandas/tools/tests/test_sjoin.py:367]: Shouldn't `if predicate == "within"` stay?

**Review by martinfleis — COMMENTED** [review]
> For reasons I don't quite understand I had to lower fiona to 1.8.18 (see this failure referencing orc and protobuf). Without forcing pyarrow=8, I was able to get an environment with fiona 1.8.19 to resolve, but that ended up installing pyarrow 3, which fails our tests. For now I have only lowered it in the ci, and not in docs/ other places.

That may be caused by something wrong on conda defaults channel or somewhere else. I wouldn't worry about that too much.

Can you add a changelog note? Otherwise I think this can be merged.

File: geopandas/tools/tests/test_sjoin.py (hunk 1 above)
Comment (m-richards) [reply geopandas/tools/tests/test_sjoin.py:367]: yep, that's an error in refactoring, sorting only makes a difference for within, redundante for other predicates.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+
+Notes on (optional) dependencies:
+
+- GeoPandas 0.13 drops support pandas 1.05 (the minimum supported
```
Comment (m-richards) [inline CHANGELOG.md:6]:
```suggestion
- GeoPandas 0.13 drops support pandas 1.0.5 (the minimum supported
```

**Review by martinfleis — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,12 +11,21 @@ Deprecations and compatibility notes:
 - Added warning that ``unary_union`` will return ``'GEOMETRYCOLLECTION EMPTY'`` instead
   of None for all-None GeoSeries. (#2618)
 
+
```
Comment (martinfleis) [inline CHANGELOG.md:14]:
```suggestion
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -11,12 +11,21 @@ Deprecations and compatibility notes:
...
 Bug fixes:
 
 - Ensure that GeoDataFrame created from DataFrame is a copy, not a view (#2667)
 - Fix mismatch between geometries and colors in ``plot()`` if an empty or missing
   geometry is present (#2224)
 
+Notes on (optional) dependencies:
```
Comment (martinfleis) [inline CHANGELOG.md:21]: Shouldn't this be merged with `Deprecations and compatibility notes` above?

Comment (m-richards) [reply CHANGELOG.md:21]: I don't mind, was just being consistent with previous changelogs which all split this out separately.

Comment (martinfleis) [reply CHANGELOG.md:20]: okay

**Comment by m-richards** [comment]
Ci is now finally passing on this (I finally realised that I missed updating the postgis section of the tests yaml) so is hopefully ready to merge.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks a lot for this!

**Pull request merged**